platform/ios/src/UIViewController+MGLAdditions.m

```diff
@@ -0,0 +1,27 @@
+#import "UIViewController+MGLAdditions.h"
+
+@implementation UIViewController (MGLAdditions)
+
+- (UIViewController *)mgl_topMostViewController
+{
+    return [self mgl_topMostViewControllerInHierarchyWith:self];
+}
+
+- (instancetype)mgl_topMostViewControllerInHierarchyWith:(UIViewController *)viewController
+{
+    if ([viewController isKindOfClass:[UINavigationController class]])
+    {
+        return [self mgl_topMostViewControllerInHierarchyWith:[(UINavigationController *)viewController visibleViewController]];
```

Comment:
Instead of passing in visibleViewController, call mgl_topMostViewController on visibleViewController. This method doesn't need to take any parameters.